app/models/concerns/call_listable.rb

```diff
@@ -50,9 +50,10 @@ def ordered_patients(line)
   TIME_BEFORE_INACTIVE = 2.weeks
 
   def clean_call_list_between_shifts
-    if last_sign_in_at.present? &&
-       last_sign_in_at < Time.zone.now - TIME_BEFORE_INACTIVE
-      patients.clear
+    last_activity = [last_sign_in_at, current_sign_in_at].reject(&:nil?).max
```

Comment:
From chat:
> It turns out that `current_sign_in_at` is the most recent sign in timestamp, and `last_sign_in_at` is the timestamp before that
> So on login, `current_sign_in_at` becomes now, and `last_sign_in_at` becomes whatever `current_sign_in_at` was

Can we just look at `current_sign_in_at`?

Reply:
Also, since these field names are unintuitive and we can't change them since they're devise, maybe a brief explainer comment?